kola/tests/ignition/mount.go

```diff
@@ -0,0 +1,411 @@
+					{
+						"name": "var-lib-containers.mount",
+						"enabled": true,
+						"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/CONTR\nWhere=/var/lib/containers\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+					},
+					{
+						"name": "var-log.mount",
+						"enabled": true,
+						"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/LOG\nWhere=/var/log\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+					}]
+			}
+		}`),
+		UserDataV3: conf.Ignition(`{
+		"ignition": {"version": "3.0.0"},
+		"storage": {
+			"disks": [
+				{
+					"device": "/dev/vda",
+					"wipeTable": false,
+					"partitions": [
+					{
+						"label": "CONTR",
+						"sizeMiB": 1024,
+						"startMiB": 0,
+						"wipePartitionEntry": true
+					},
+					{
+						"label": "LOG",
+						"sizeMiB": 1024,
+						"startMiB": 0,
+						"wipePartitionEntry": true
+					}
+					]
+				}
+			],
+			"filesystems": [
+				{
+					"path": "/var/lib/containers",
+					"device": "/dev/disk/by-partlabel/CONTR",
+					"format": "xfs",
+					"label": "CONTR",
+					"wipeFilesystem": true
+				},
+				{
+					"path": "/var/log",
+					"device": "/dev/disk/by-partlabel/LOG",
+					"format": "xfs",
+					"label": "LOG",
+					"wipeFilesystem": true
+				}
+			],
+			"files": [
+				{
+					"path": "/var/lib/containers/hello.txt",
+					"contents": {
+						"source": "data:,hello%20world%0A"
+					}
+				},
+				{
+					"path": "/var/log/hello.txt",
+					"contents": {
+						"source": "data:,hello%20world%0A"
+					}
+				}
+			]
+		},
+		"systemd": {
+			"units": [
+				{
+					"name": "var-lib-containers.mount",
+					"enabled": true,
+					"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/CONTR\nWhere=/var/lib/containers\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+				},
+				{
+					"name": "var-log.mount",
+					"enabled": true,
+					"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/LOG\nWhere=/var/log\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+				}]
+		}
+	}`),
+		Distros: []string{"fcos", "rhcos"},
+	})
+}
+
+func mountVarLibContainers(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	out := c.MustSSH(m, "cat /proc/mounts")
+	if !strings.Contains(string(out), "/var/lib/containers") {
+		c.Fatalf("No partition mounted to /var/lib/containers")
+	}
+
+	out = c.MustSSH(m, "cat /var/lib/containers/hello.txt")
+	if string(out) != "hello world" {
+		c.Fatalf("Failed to write content to /var/lib/containers/hello.txt")
+	}
+}
+
+func mountVarLog(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	out := c.MustSSH(m, "cat /proc/mounts")
+	if !strings.Contains(string(out), "/var/log") {
+		c.Fatalf("No partition mounted to /var/log")
+	}
+
+	out = c.MustSSH(m, "cat /var/log/hello.txt")
+	if string(out) != "hello world" {
+		c.Fatalf("Failed to write content to /var/log/hello.txt")
+	}
+}
+
+func mountMultiple(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	out := c.MustSSH(m, "cat /proc/mounts")
```

Comment:
With the way that you've written these tests the validation could be split out into a sub-function to reduce duplication, e.x.:
```golang
func mountValidate(c cluster.TestCluster, m platform.Machine, mountContents, path string) {
    if strings.Contains(mountContents, path) {
        c.Fatalf("No partition mounted to %s", path)
    }

    fPath := filepath.Join(path, "hello.txt")
    fileContents := c.MustSSH(m, fPath)
    if string(fileContents) != "hello world" {
        c.Fatalf("Failed to write content to %s", fPath)
    }
}

func mountMultiple(c cluster.TestCluster) {
    m := c.Machines()[0]
    mountContents := c.MustSSH(m, "cat /proc/mounts")
    mountValidate(c, m, mountContents, "/var/lib/containers")
    mountValidate(c, m, mountContents, "/var/lib/log")
}
```

Reply:
Thanks for sharing the code! I've updated with helper functions based on you comment.